src/uu/factor/src/factor.rs

```diff
@@ -18,139 +18,94 @@ extern crate rand;
 #[macro_use]
 extern crate uucore;
 
-use numeric::*;
-use rand::distributions::{Distribution, Uniform};
-use rand::rngs::SmallRng;
-use rand::{thread_rng, SeedableRng};
-use std::cmp::{max, min};
+use std::collections::HashMap;
+use std::fmt;
 use std::io::{stdin, BufRead};
-use std::mem::swap;
-use std::num::Wrapping;
+use std::ops;
 
+mod miller_rabin;
 mod numeric;
-
-include!(concat!(env!("OUT_DIR"), "/prime_table.rs"));
+mod rho;
+mod table;
 
 static SYNTAX: &str = "[OPTION] [NUMBER]...";
 static SUMMARY: &str = "Print the prime factors of the given number(s).
  If none are specified, read from standard input.";
 static LONG_HELP: &str = "";
 
-fn rho_pollard_pseudorandom_function(x: u64, a: u64, b: u64, num: u64) -> u64 {
-    if num < 1 << 63 {
-        (sm_mul(a, sm_mul(x, x, num), num) + b) % num
-    } else {
-        big_add(big_mul(a, big_mul(x, x, num), num), b, num)
-    }
+struct Factors {
+    f: HashMap<u64, u8>,
 }
 
-fn gcd(mut a: u64, mut b: u64) -> u64 {
-    while b > 0 {
-        a %= b;
-        swap(&mut a, &mut b);
+impl Factors {
+    fn new() -> Factors {
+        Factors { f: HashMap::new() }
+    }
+
+    fn add(&mut self, prime: u64, exp: u8) {
+        assert!(exp > 0);
+        self.f.insert(prime, exp + self.f.get(&prime).unwrap_or(&0));
```

Comment:
This fails on Rust versions that do not have NLL, and my brain is a bit too cooked and full of number theory right now to fix it, so feel free to provide a fix  ;)

Reply:
```suggestion
        let n = self.f.get(&prime).unwrap_or(&0).clone();
        self.f.insert(prime, exp + n);
```
This compiles and tests without error on v1.31.0+. Since it's a `u8` clone, it should be effectively optimized away by the compiler and shouldn't affect the efficiency/speed of the code.

Reply:
... edited to appease `cargo clippy`.

Reply:
@rivy Thanks!  I thought I had tried that, but either I had a typo somewhere, or I had another uncommited changed which was broken and mistakenly blamed the no-NLL fix.